extensions/amp-viewer-integration/0.1/messaging.js

```diff
@@ -66,123 +86,174 @@ export class Messaging {
   handleMessage_(event) {
     if (!event || event.source != this.target_ ||
       event.origin != this.targetOrigin_) {
+      this.logError_(TAG +
```

Comment:
I just don't see these conditions as errors. IMO these are totally expected cases in the real world.